### ISSUE TITLE
Implement profile mode for desktop shells

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#33203]
For consistency with the other platforms and with `flutter` tool behaviors, we should presumably implement the Profile mode for the desktop shells.

See also #30672

**Comment by jonahwilliams** [comment]
I think profile mode more or less "just works" provided you have AOT support and an uploaded host_profile artifact.

**Comment by jonahwilliams** [comment]
I think profile mode more or less "just works" provided you have AOT support and an uploaded host_profile artifact.

**Comment by stuartmorgan** [comment]
If nothing else though it'll require some plumbing in the native projects to make it a configuration that passes the right flags.

**Comment by Nipodemos** [comment]
hello
i think this issue should not be closed, since *i think* flutter still not have profile mode for windows and linux.
Sorry if i got something wrong

**Comment by stuartmorgan** [comment]
You are correct, thanks. I forgot that I didn't split this apart into per-platform bugs when I split Release (see https://github.com/flutter/flutter/issues/30672#issuecomment-520996422 for reasoning).

Filed #39664 and #39665, and I'll retroactively rescope this to just macOS.
